File: plugins/node/template/addons/queue/src/workers/example.worker.ts
import { Worker, Job } from "bullmq";
import { env } from "../config/env";

const connection = { url: env.REDIS_URL };

export interface ExampleJobData {
  message: string;
  userId?: number;
}

export const exampleWorker = new Worker<ExampleJobData>(
  "example",
  async (job: Job<ExampleJobData>) => {
    console.log(`[Worker] Processing job ${job.id}: ${job.data.message}`);
    // TODO: replace with real job logic
    await new Promise((res) => setTimeout(res, 100));
    return { processed: true };
  },
  { connection },
);

exampleWorker.on("completed", (job) => {
  console.log(`[Worker] Job ${job.id} completed`);
});

exampleWorker.on("failed", (job, err) => {
  console.error(`[Worker] Job ${job?.id} failed: ${err.message}`);
});

File: plugins/python/template/addons/queue/app/workers/example_worker.py
import asyncio

from app.core.logging import get_logger

logger = get_logger(__name__)


async def example_job(ctx: dict, message: str, user_id: int | None = None) -> dict:
    """Example background job — replace with real logic."""
    logger.info(f"Processing job: message={message!r} user_id={user_id}")
    await asyncio.sleep(0.1)  # simulate work
    return {"processed": True, "message": message}


# arq worker settings — imported by arq CLI as WorkerSettings
class WorkerSettings:
    functions = [example_job]
    # Adjust Redis URL via REDIS_URL env var loaded by arq from the env
    # To run: arq app.workers.example_worker.WorkerSettings
